@Carreau -- I'm finally able to get back to this, but am confused by what I see wrt these failing tests.

The build log shows that the first failure is in `core/tests/test_completer.py`. However I cannot reproduce this failure as all test are passing for me locally:

```
❯ git branch
* feature/add-line-numbers-to-prompt
  main
❯ git pull
Already up to date.
❯ pytest core/tests/test_completer.py
/Users/chamill/git/forks/ipython/ipython/env/lib/python3.12/site-packages/pytest_asyncio/plugin.py:173: DeprecationWarning: You're using an outdated version of pytest. Newer releases of pytest-asyncio will not be compatible with this pytest version. Please update pytest to version 7 or later.
  warnings.warn(
========================================================= test session starts =========================================================
platform darwin -- Python 3.12.0, pytest-6.2.5, py-1.11.0, pluggy-1.3.0
rootdir: /Users/chamill/git/forks/ipython/ipython, configfile: pyproject.toml
plugins: asyncio-0.20.3, anyio-4.1.0
asyncio: mode=Mode.STRICT
collected 88 items

core/tests/test_completer.py ...............X........................................................................           [100%]

========================================================== warnings summary ===========================================================
../env/lib/python3.12/site-packages/dateutil/tz/tz.py:37
  /Users/chamill/git/forks/ipython/ipython/env/lib/python3.12/site-packages/dateutil/tz/tz.py:37: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    EPOCH = datetime.datetime.utcfromtimestamp(0)

IPython/core/tests/test_completer.py: 65 warnings
  /Users/chamill/git/forks/ipython/ipython/env/lib/python3.12/unittest/case.py:580: RuntimeWarning: TestResult has no addDuration method
    warnings.warn("TestResult has no addDuration method",

-- Docs: https://docs.pytest.org/en/stable/warnings.html
======================================================== slowest 10 durations =========================================================
0.61s call     IPython/core/tests/test_completer.py::test_unicode_range
0.38s call     IPython/core/tests/test_completer.py::TestCompleter::test_dict_key_completion_string
0.14s call     IPython/core/tests/test_completer.py::TestCompleter::test_dict_key_completion_unicode_py3
0.12s call     IPython/core/tests/test_completer.py::TestCompleter::test_jedi
0.07s call     IPython/core/tests/test_completer.py::TestCompleter::test_matcher_suppression_with_jedi
0.07s call     IPython/core/tests/test_completer.py::TestCompleter::test_func_kw_completions
0.07s call     IPython/core/tests/test_completer.py::TestCompleter::test_greedy_completions
0.05s call     IPython/core/tests/test_completer.py::TestCompleter::test_abspath_file_completions
0.02s call     IPython/core/tests/test_completer.py::TestCompleter::test_aimport_module_completer
0.02s call     IPython/core/tests/test_completer.py::TestCompleter::test_import_module_completer
============================================= 87 passed, 1 xpassed, 66 warnings in 1.88s ==============================================
```

In fact, out of fully 1303 tests just a single one is failing for me locally (and, incidentally, I checked and it was also failing on `main`?):

```
❯ pytest core/tests/ extensions/tests/ external/tests/ lib/tests/ terminal/tests/ testing/tests/ tests/ utils/tests/
/Users/chamill/git/forks/ipython/ipython/env/lib/python3.12/site-packages/pytest_asyncio/plugin.py:173: DeprecationWarning: You're using an outdated version of pytest. Newer releases of pytest-asyncio will not be compatible with this pytest version. Please update pytest to version 7 or later.
  warnings.warn(
========================================================= test session starts =========================================================
platform darwin -- Python 3.12.0, pytest-6.2.5, py-1.11.0, pluggy-1.3.0
rootdir: /Users/chamill/git/forks/ipython/ipython, configfile: pyproject.toml
plugins: asyncio-0.20.3, anyio-4.1.0
asyncio: mode=Mode.STRICT
collected 1331 items

core/tests/test_alias.py ....                                                                                                   [  0%]
core/tests/test_application.py ...                                                                                              [  0%]
core/tests/test_async_helpers.py ..s........                                                                                    [  1%]
core/tests/test_autocall.py ..                                                                                                  [  1%]
core/tests/test_compilerop.py ......                                                                                            [  1%]
core/tests/test_completer.py ...............X........................................................................           [  8%]
core/tests/test_completerlib.py ............                                                                                    [  9%]
core/tests/test_debugger.py ........ss..                                                                                        [ 10%]
core/tests/test_display.py .........................                                                                            [ 12%]
core/tests/test_displayhook.py .......                                                                                          [ 12%]
core/tests/test_events.py .....                                                                                                 [ 13%]
core/tests/test_exceptiongroup_tb.py .........                                                                                  [ 13%]
core/tests/test_extension.py ...                                                                                                [ 14%]
core/tests/test_formatters.py ..............................                                                                    [ 16%]
core/tests/test_guarded_eval.py ............................................................................................... [ 23%]
............................................................................................................................... [ 32%]
.                                                                                                                               [ 33%]
core/tests/test_handlers.py .                                                                                                   [ 33%]
core/tests/test_history.py .........                                                                                            [ 33%]
core/tests/test_hooks.py ...                                                                                                    [ 34%]
core/tests/test_imports.py .................                                                                                    [ 35%]
core/tests/test_inputsplitter.py ........................................................................                       [ 40%]
core/tests/test_inputtransformer.py .................                                                                           [ 41%]
core/tests/test_inputtransformer2.py .........................................                                                  [ 45%]
core/tests/test_inputtransformer2_line.py ......                                                                                [ 45%]
core/tests/test_interactiveshell.py ......................................................................................      [ 51%]
core/tests/test_iplib.py .......                                                                                                [ 52%]
core/tests/test_logger.py ..                                                                                                    [ 52%]
core/tests/test_magic.py ..........................................................F........................................... [ 60%]
...                                                                                                                             [ 60%]
core/tests/test_magic_arguments.py .                                                                                            [ 60%]
core/tests/test_magic_terminal.py ...........                                                                                   [ 61%]
core/tests/test_oinspect.py ..................................                                                                  [ 64%]
core/tests/test_page.py .                                                                                                       [ 64%]
core/tests/test_paths.py ...........                                                                                            [ 64%]
core/tests/test_prefilter.py ......                                                                                             [ 65%]
core/tests/test_profile.py .....                                                                                                [ 65%]
core/tests/test_prompts.py ...                                                                                                  [ 65%]
core/tests/test_pylabtools.py ..............                                                                                    [ 67%]
core/tests/test_run.py .....s...............................                                                                    [ 69%]
core/tests/test_shellapp.py ...                                                                                                 [ 70%]
core/tests/test_splitinput.py ..                                                                                                [ 70%]
core/tests/test_ultratb.py .......................                                                                              [ 71%]
extensions/tests/test_autoreload.py ..........                                                                                  [ 72%]
extensions/tests/test_storemagic.py ..                                                                                          [ 72%]
external/tests/test_qt_loaders.py .                                                                                             [ 72%]
lib/tests/test_backgroundjobs.py ....                                                                                           [ 73%]
lib/tests/test_clipboard.py .                                                                                                   [ 73%]
lib/tests/test_deepreload.py ...                                                                                                [ 73%]
lib/tests/test_display.py ............................                                                                          [ 75%]
lib/tests/test_editorhooks.py .                                                                                                 [ 75%]
lib/tests/test_imports.py ...                                                                                                   [ 75%]
lib/tests/test_latextools.py ..ss........s.                                                                                     [ 76%]
lib/tests/test_lexers.py .                                                                                                      [ 77%]
lib/tests/test_pretty.py ...................................                                                                    [ 79%]
lib/tests/test_pygments.py ...                                                                                                  [ 79%]
terminal/tests/test_debug_magic.py .                                                                                            [ 79%]
terminal/tests/test_embed.py ..                                                                                                 [ 80%]
terminal/tests/test_help.py .......                                                                                             [ 80%]
terminal/tests/test_interactivshell.py .........                                                                                [ 81%]
terminal/tests/test_pt_inputhooks.py s                                                                                          [ 81%]
terminal/tests/test_shortcuts.py ........................................................                                       [ 85%]
testing/tests/test_decorators.py .ss..s.ss....s                                                                                 [ 86%]
testing/tests/test_ipunittest.py .....                                                                                          [ 87%]
testing/tests/test_tools.py .s........                                                                                          [ 87%]
utils/tests/test_capture.py ........................                                                                            [ 89%]
utils/tests/test_decorators.py .                                                                                                [ 89%]
utils/tests/test_deprecated.py .                                                                                                [ 89%]
utils/tests/test_dir2.py ....                                                                                                   [ 90%]
utils/tests/test_imports.py ......                                                                                              [ 90%]
utils/tests/test_importstring.py ...                                                                                            [ 90%]
utils/tests/test_io.py ...                                                                                                      [ 90%]
utils/tests/test_module_paths.py ......                                                                                         [ 91%]
utils/tests/test_openpy.py ...                                                                                                  [ 91%]
utils/tests/test_path.py ss...s.....s.....s...........s                                                                         [ 93%]
utils/tests/test_process.py .s.....ssss........                                                                                 [ 95%]
utils/tests/test_pycolorize.py ........                                                                                         [ 95%]
utils/tests/test_shimmodule.py ..                                                                                               [ 96%]
utils/tests/test_sysinfo.py ..                                                                                                  [ 96%]
utils/tests/test_tempdir.py ..                                                                                                  [ 96%]
utils/tests/test_text.py ........................                                                                               [ 98%]
utils/tests/test_tokenutil.py ...................                                                                               [ 99%]
utils/tests/test_wildcard.py ......                                                                                             [100%]

============================================================== FAILURES ===============================================================
___________________________________________________________ test_extension ____________________________________________________________

    def test_extension():
        # Debugging information for failures of this test
        print('sys.path:')
        for p in sys.path:
            print(' ', p)
        print('CWD', os.getcwd())

>       pytest.raises(ImportError, _ip.magic, "load_ext daft_extension")
E       Failed: DID NOT RAISE <class 'ImportError'>

core/tests/test_magic.py:833: Failed

[...]

======================================================= short test summary info =======================================================
FAILED core/tests/test_magic.py::test_extension - Failed: DID NOT RAISE <class 'ImportError'>
================================ 1 failed, 1303 passed, 26 skipped, 1 xpassed, 454 warnings in 46.25s =================================

```

Do you have any insights? This is my first exposure to GitHub builds so perhaps there are variables I'm unaware of when it comes to reproducibility...

For context, all outputs above were generated with a fresh clone of the repository created using [the same commands from my earlier comment, though thanks to you they now seem to work](https://github.com/ipython/ipython/pull/14223#issuecomment-1785794024).

--- a/IPython/terminal/interactiveshell.py
+++ b/IPython/terminal/interactiveshell.py
@@ -747,7 +747,7 @@ class TerminalInteractiveShell(InteractiveShell):
         def get_message():
             return PygmentsTokens(self.prompts.in_prompt_tokens())
 
-        if self.editing_mode == 'emacs':
+        if self.editing_mode == 'emacs' and self.prompt_line_number_format == '':
             # with emacs mode the prompt is (usually) static, so we call only
             # the function once. With VI mode it can toggle between [ins] and
             # [nor] so we can't precompute.

